src/controllers/csi/provisioner/agent.go

```diff
@@ -78,7 +99,7 @@ func getUrlProperties(version string) *url.Properties {
 	}
 }
 
-func setupImageInstaller(ctx context.Context, fs afero.Fs, apiReader client.Reader, certPath string, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) {
+func setupImageInstaller(ctx context.Context, fs afero.Fs, pathResolver metadata.PathResolver, apiReader client.Reader, certPath, digest string, dynakube *dynatracev1beta1.DynaKube) (installer.Installer, error) {
```

Comment:
Those are way too many parameters and must be refactored. 
Maybe replace it using the builder pattern.